datumaro/components/progress_reporting.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from __future__ import annotations
+
+from typing import ContextManager, Iterable, Optional, Tuple, TypeVar
+import math
+
+T = TypeVar('T')
+
+class ProgressReporter(ContextManager['ProgressReporter']):
+    """
+    Only one set of methods must be called:
+    - init - report_status - close
+    - iter - close
+    - split
+
+    Must be close()-d after use, can be used as a context manager.
+    Must not be reused after closing.
+    """
+
+    def close(self):
+        """Closes the progress bar"""
+        raise NotImplementedError
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type=None, exc_value=None, traceback=None):
+        self.close()
+
+    @property
+    def frequency(self) -> float:
+        """
+        Returns reporting frequency.
+
+        For example, 0.1 would mean every 10%.
```

Comment:
Doesn't this actually make it the _period_, not the frequency?

Reply:
Yes, probably